api/src/main/java/org/hawkular/apm/api/utils/PropertyUtil.java

```diff
@@ -191,12 +194,14 @@ public static Integer getPropertyAsInteger(String name, Integer def) {
      *
      * @return The property as a boolean, or false if not found
      */
-    public static boolean getPropertyAsBoolean(String name) {
+    public static Boolean getPropertyAsBoolean(String name) {
```

Comment:
I'm puzzled by this change.

Why would you want to return `Boolean` instead of `boolean`? All your current consumers are now auto unboxing it into `boolean`, as I don't see related changes on this PR. So, you might get better performance on most (all?) JVMs by auto unboxing everything here.

If you are returning `null` if `value == null`, you are effectively pushing a `NullPointerException` to your calling code, as they expect a `boolean`, but auto unboxing will not convert `null` into a valid boolean, throwing a NPE instead. So, it's probably better to throw it here and be explicit about your requirements.

Even if you have clients expecting `Boolean` instead of `boolean`, you are now effectively saying: I'm returning a "true", "false" or "null". As a consumer, I would not expect that, even if the method returns a `Boolean`.